process/factory/containers/interceptorsContainer.go

```diff
@@ -123,6 +125,22 @@ func (ic *interceptorsContainer) Iterate(handler func(key string, interceptor pr
 	}
 }
 
+// Close will call the close method on all contained interceptors
+func (ic *interceptorsContainer) Close() error {
+	var errFound error
+	ic.Iterate(func(key string, interceptor process.Interceptor) bool {
```

Comment:
using the generic mutex map seems to be 5 times slower for the iterating purpose because of the needed casts and getters.

``` GOLANG
func BenchmarkMutexMap(b *testing.B) {
	mm := NewMutexMap()

	for i := 0; i < 100; i++ {
		mm.Insert(fmt.Sprintf("key-%d", i), "value")
	}

	for i := 0; i < b.N; i++ {
		keys := mm.Keys()
		for _, keyVal := range keys {
			key, ok := keyVal.(string)
			if !ok {
				continue
			}

			_, ok = mm.Get(key)
			if !ok {
				continue
			}
		}
	}
}

func BenchmarkFixedTypeMap(b *testing.B) {
	mm := make(map[string]string)
	mt := sync.RWMutex{}

	for i := 0; i < 100; i++ {
		mt.Lock()
		mm[fmt.Sprintf("key-%d", i)] = "value"
		mt.Unlock()
	}

	for i := 0; i < b.N; i++ {
		mt.RLock()
		for range mm {

		}
		mt.RUnlock()
	}
}
```
```
// BenchmarkMutexMap-12    	  220370	      5368 ns/op
// BenchmarkFixedTypeMap-12    	 1000000	      1016 ns/op
```

It can remain the same because there aren't many interceptors, but for other use cases I'd say we should use a specific map with a specific mutex

Reply:
interesting comment :+1: . Will add a TODO to change the implementations. We have have several implementations like this. Added task EN-10283

Reply:
Will remain as it is because of the duplicated code reduction reason. Would be fixed when we will migrate to the golang variant that will support generics.